aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -104,13 +109,35 @@ public GlueCatalog() {
 
   @Override
   public void initialize(String name, Map<String, String> properties) {
+    AwsClientFactory awsClientFactory;
+    FileIO catalogFileIO;
+    if (PropertyUtil.propertyAsBoolean(
+        properties,
+        AwsProperties.GLUE_LAKEFORMATION_ENABLED,
+        AwsProperties.GLUE_LAKEFORMATION_ENABLED_DEFAULT)) {
+      String factoryImpl = PropertyUtil.propertyAsString(properties, AwsProperties.CLIENT_FACTORY, null);
+      ImmutableMap.Builder<String, String> builder = ImmutableMap.<String, String>builder().putAll(properties);
+      if (factoryImpl == null) {
+        builder.put(AwsProperties.CLIENT_FACTORY, LakeFormationAwsClientFactory.class.getName());
+      }
```

Comment:
can we just assign `LakeFormationAwsClientFactory.class.getName()` as the defaultValue 

```java
      String factoryImpl = PropertyUtil.propertyAsString(properties, AwsProperties.CLIENT_FACTORY, LakeFormationAwsClientFactory.class.getName());
      ImmutableMap.Builder<String, String> builder = ImmutableMap.<String, String>builder().putAll(properties);
      builder.put(AwsProperties.CLIENT_FACTORY, LakeFormationAwsClientFactory.class.getName());
```

in the log anyhow we are doing the instance of check on `AwsClientFactories.from(catalogProperties);`


Am I missing something here ?

Reply:
Immutable map cannot assign duplicated values